doc/changelog/11-standard-library/18032-warn_vector.rst

```diff
@@ -0,0 +1,7 @@
+- **Added:**
+  A warning on :g:`Vector.t` to make its new users aware of the inconvenience
+  of this dependently typed representation of fixed-length lists. This is not
+  a deprecation and there is no intent to remove it from the standard library.
+  Use option `-w -deprecated-since-stdlib-vector` to silence the warning
```

Comment:
It's indeed not ideal but the name has to be `deprecated-since-something` for technical reasons (deprecations are the tool we currently have to display a warning).